### PR TITLE
Deps: Update Jest to 22

### PR DESCRIPTION
**Pull request by sirreal**
Updates [Jest to v22](https://facebook.github.io/jest/blog/2017/12/18/jest-22.html).

This PR also gets us off of a beta release of `jest-docblock` 👍 

Thanks @gziolo for the nudge https://github.com/WordPress/gutenberg/pull/4073#pullrequestreview-84362557

## Notable improvements:
* The mess of `requestAnimationFrame` warnings shown during tests [are gone](https://facebook.github.io/jest/blog/2017/12/18/jest-22.html#good-bye-node-4-welcome-jsdom-11) 🎉 
* [Codeframe](https://facebook.github.io/jest/blog/2017/12/18/jest-22.html#codeframe-in-test-failures) is 💯  
* `jest-docblock` out of beta
* [More](https://facebook.github.io/jest/blog/2017/12/18/jest-22.html)

## Follow up
* Worth looking at the new [`jest-runner-eslint`](http://facebook.github.io/jest/blog/#custom-runners-easy-parallelization-with-jest-worker)

## Testing
1. Are tests passing? 😁

### CODE REVIEW

**Comment by gziolo** [comment]
CircleCi doesn't look as happy as we do :D

**Comment by sirreal** [comment]
> CircleCi doesn't look as happy as we do :D

Nope, not ready yet, still in progress.

Only 2 failures though 😁

File: client/lib/accept/test/index.js [hunk 1]
```diff
@@ -48,15 +47,8 @@ describe( '#accept()', () => {
...
-			process.nextTick( function() {
-				expect( document.querySelector( '.accept-dialog' ) ).to.be.null;
-			} );
-			jest.runAllTimers();
-			jest.useRealTimers();
-		} );
-
+		accept( 'Are you sure?', () => {} );
 		document.querySelector( '.button.is-primary' ).click();
+		expect( document.querySelector( '.accept-dialog' ) ).to.be.null;
```
Comment (sirreal) [inline client/lib/accept/test/index.js]: It was surprisingly difficult to come to this simple solution this test fix.

Comment (gziolo) [reply client/lib/accept/test/index.js]: Lol, should work as expected when you compare with one of the previous tests :)

**Comment by gziolo** [comment]
There are some strange errors coming from `eslines`. Did you investigate them?

**Comment by sirreal** [comment]
> There are some strange errors coming from eslines. Did you investigate them?

Those were a surprise, yes I've started investigating.

**Comment by sirreal** [comment]
The upgraded `eslint-plugin-jest` was incompatible with the versions of some other eslint related packages we're currently using. Rather than enter into dependency-upgrade-hell, I've rolled back and now CI is 💚 

Ready for review!

**Comment by gziolo** [comment]
So they bumped version to 22.0.3, interesting :)

**Comment by sirreal** [comment]
I've just tried the new [experimental `--detectLeaks`](https://facebook.github.io/jest/blog/2017/12/18/jest-22.html#experimental-leak-detection):

```sh
npm run test-client -- --detectLeaks
```

I expected some unusual behavior, but it fails for _every single test_, both client and server. I'm curious what we're doing to cause that 🤔

**Comment by gziolo** [comment]
Let me know if you discover what is causing those leaks. It might be the fact that we disable network traffic in here: https://github.com/Automattic/wp-calypso/blob/master/test/client/setup-test-framework.js#L11.

You can also try to compare with Gutenberg which has a slightly different setup.

**Pull request merged**